src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -157,7 +159,7 @@ def validate_client_parameters(cmd, namespace):
     # otherwise, simply try to retrieve the remaining variables from environment variables
     if not n.account_name:
         n.account_name = get_config_value(cmd, 'storage', 'account', None)
-    if not n.account_key:
+    if not n.account_key and not n.sas_token:
         n.account_key = get_config_value(cmd, 'storage', 'key', None)
     if not n.sas_token:
```

Comment:
```suggestion
    if not n.account_key and not n.sas_token:
```
If `account_key` has been retrieved from enviroment variable, then we can omit retrieve `sas_token`, right? But this will make our code ugly😂

Reply:
we should not omit sas token although we could get account key from config value. Because user input is first priority.